examples/hitl/rearrange_v2/rearrange_v2.py

```diff
@@ -139,6 +140,21 @@ def __init__(self, app_service):
         self._task_instruction = ""
         self._data_logger = DataLogger(app_service=self._app_service)
 
+        if self._app_service.hitl_config.networking.enable:
+            self._app_service.remote_client_state.on_client_connected.registerCallback(
+                self._on_client_connected
+            )
+            self._app_service.remote_client_state.on_client_disconnected.registerCallback(
+                self._on_client_disconnected
+            )
+            self._paused = True
+
+    def _on_client_connected(self, connection: ConnectionRecord):
+        self._paused = False
+
+    def _on_client_disconnected(self, disconnection: DisconnectionRecord):
+        self._paused = True
```

Comment:
Does this apply to multi-user applications?
If so, does this pause when either client disconnects, or both?

Reply:
Kinda.

The application is always on. The pausing logic is useful to prevent the server from stepping, and to prevent the client to download the previously loaded scene upon connecting.

This iteration of `rearrange_v2` is only single user. Multiplayer support is not merged yet. In multplayer, application state will be controlled by a small state machine. The app will only be paused if nobody is connected.